lib/spack/spack/mirror.py

```diff
@@ -202,57 +213,26 @@ def create(path, specs, **kwargs):
         'error': []
     }
 
-    # Iterate through packages and download all safe tarballs for each
-    for spec in version_specs:
-        add_single_spec(spec, mirror_root, categories, **kwargs)
+    mirror_cache = spack.caches.MirrorCache(mirror_root)
+    try:
+        spack.caches.mirror_cache = mirror_cache
+        # Iterate through packages and download all safe tarballs for each
+        for spec in version_specs:
+            add_single_spec(spec, mirror_root, categories, **kwargs)
+    finally:
+        spack.caches.mirror_cache = None
```

Comment:
This try/finally is the part that makes me think that the mirror cache should be moved out of `spack.caches`.  I think it should be integrated somehow with the logic for `FsCache`, but I think factoring all of that properly is a larger task.  There's a lot of global/module-scope state in the fetch and package build logic right now that I would like to get rid of -- this adds to it b/c `mirror_cache` is basically a global here.